FranciscoPerez/react-challenge-project-jan-2023
# Fp 7 update dependencies

## Changes
1. Update all installed packages to latest versions.
2. Refactor `appRouter.js` to use the React Router v6 implementation.
3. Refactor `\src\index.js` to use the React 18 root rendering implementation.
4. Change `login.js` to a functional React component and implement `useNavigate` hook to programmatically redirect user to `\view-orders` page upon login.

## Purpose
Using outdated packages make your project susceptible to security vulnerabilities and deprecation warnings/issues.

## Approach
Update each package to the latest version using `npm install <package-name>@latest`.

[Issue #7](https://github.com/Shift3/react-challenge-project-jan-2023/issues/7)

## Testing Steps
1. In the `/login` page, enter password/email and click the login button.
2. In the `/order` page, enter menu item and quantity then click the `Order It!` button.
3. In the `/view-orders` page, verify previously created orders are present.
4. Click the Logout button and verify that you are redirected to the `/login` page.

## Learning
I referenced the official npm website for information on the `npm install` command.

[npm install Documentation](https://docs.npmjs.com/cli/v8/commands/npm-install)

I referenced the following blog article below that covered other npm features such as `npm outdated` to view all outdated packages.

[Upgrading npm Dependencies](https://www.carlrippon.com/upgrading-npm-dependencies/)

I also referenced a migration guide for React 18 to assist in refactoring outdated features.

[React 18 Migration Guide](https://react.dev/blog/2022/03/08/react-18-upgrade-guide)

Closes #7
